### PR TITLE
[CI] Change test report

### PR DESCRIPTION
**Pull request by talregev**
Change test report.
I made a mistake and added the test report badge to readme. This badge don't indicate if the tests are past. it just indicate if it succeeded to create the job for it. It misleading. 
So I implement it differently, more naturally. The test report now wait for the job with all matrix option, then download the artifact and create the report similar to before. But now it on the same workflow. 
The report still appear on random workflow. It an know bug that developer of the reporter need better api from github action. 
After this PR there is no need for extra workflow.

### CODE REVIEW

File: .github/workflows/android.yml [hunk 1]
```diff
@@ -143,3 +143,21 @@ jobs:
         with:
           fail_ci_if_error: true
           verbose: false
+  report:
+    needs: build
+    runs-on: ubuntu-latest
```
Comment (AenBleidd) [inline .github/workflows/android.yml:148]: Do we really need to run this report on a separate build flow (build machine to be exact) and can't just run it in the same flow?

Comment (talregev) [reply .github/workflows/android.yml:148]: I think there is something connected to permission. That why it needed to split to 2 workflow, as suggested in the readme of the dev github test reporter.

**Review by AenBleidd — APPROVED** [review]
Looks good to me

**Comment by talregev** [comment]
> Looks good to me

I test it more before merge.

**Comment by AenBleidd** [comment]
> I test it more before merge.

Yes please, because currently looks like it's failing trying to create report

File: .github/workflows/android.yml (hunk 1 above)
Comment (talregev) [reply .github/workflows/android.yml:148]: That why it cannot be on the same workflow, even as I did here:
`Resource not accessible by integration`

**Comment by talregev** [comment]
> > I test it more before merge.
> 
> Yes please, because currently looks like it's failing trying to create report

I will create another PR just delete the misleading badge.

**Pull request closed without merging**